src/openapi/streaming/connection/connection.js

```diff
@@ -0,0 +1,196 @@
+import log from './../../../log';
+import WebsocketTransport from './transport/websocket-transport';
+import SignalrTransport from './transport/signalr-transport';
+
+const LOG_AREA = 'PlainWebSocketsTransport';
+const DEFAULT_TRANSPORTS = [WebsocketTransport, SignalrTransport];
+
+const TRANSPORT_NAME_MAP = {
+    // New, plain RAW WebSocket transport.
+    [WebsocketTransport.NAME]: WebsocketTransport,
+
+    // Backward compatible mapping to SignalR.
+    'webSockets': SignalrTransport,
+    'longPolling': SignalrTransport,
+};
+
+const NOOP = () => {};
+
+const STATE_CREATED = 'connection-state-created';
+const STATE_STARTED = 'connection-state-started';
+const STATE_STOPPED = 'connection-state-stopped';
+
+function onTransportFail(error) {
+    log.error(LOG_AREA, 'Transport failed', { error });

```

Comment:
For initial release I would recommend using aggressive fallback to signalR and cover it with additional logs about fallback reasons. For example:
1. Failed due to handshake connection error response.
2. Failed due to handshake connection error.
3. Failed due to authorization failure.
4. Failed due to close with error code. 
5. Failed due to lack of supported types (Uint8Array, WebSocket, DataView).

We could analyse results and act accordingly for Phase 2.

However if you believe that current logs already paint a relatively clear picture I would suggest following. 
- If failures are due to handshake or authorization issues, it could mean connection failures. In such scenarios try to reconnect within same transport.
- If failure is due to missing support ie. lack of WebSocket class, than fallback to signalR. 

Anyway, I wouldn't recommend dropping signalr in first plain/raw websocket release. However I would also love to drop it in the end.

Reply:
What about, fully connected, getting data, then gets disconnected?

Reply:
Disconnect (onclose) should start reconnect loop. However it's highly probable that next connect attempt will fail, which in effect will call onerror and then transport will fail and cause fallback.

I like the idea of trying to reconnect more aggressively if transport worked at least once! Will implement.